app/classifier/tasks/survey/index.cjsx

```diff
@@ -84,7 +87,13 @@ module.exports = React.createClass
   handleAnnotation: (choice, answers, e) ->
     filters = JSON.parse JSON.stringify @state.filters
     value = @props.annotation.value?.slice?(0) ? []
-    value.push {choice, answers, filters}
+    # @ is undefined within the scope of findIndex
+    currentSelection = @state.selectedChoiceID
+    annotationIndex = value.findIndex (value) -> value.choice is currentSelection
```

Comment:
That's odd. [indexOf](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) should return -1 if the argument doesn't exist.